sound/soc/sof/ops.h

```diff
@@ -127,6 +129,8 @@ static inline int snd_sof_dsp_core_put(struct snd_sof_dev *sdev, int core)
 
 		/* and update enabled_cores_mask */
 		sdev->enabled_cores_mask &= ~BIT(core);
+
+		dev_dbg(sdev->dev, "Core %d powered down\n", core);
```

Comment:
this is starting to be really verbose, isn't it?

Reply:
@plbossart why do you say that? this is a snippet of what you'd see:
[ 1948.700794] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30020000: GLB_TPLG_MSG: COMP_FREE
[ 1948.701076] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x30020000: GLB_TPLG_MSG: COMP_FREE
[ 1948.701096] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
[ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x40070000: GLB_PM_MSG: CORE_ENABLE
[ 1948.701422] sof-audio-pci-intel-tgl 0000:00:1f.3: core 3 powered down
[ 1948.701429] sof-audio-pci-intel-tgl 0000:00:1f.3: widget SSP0.OUT freed

Without the above prints, all I know is something was either powered on/off but not what

Reply:
in fact I'd like to reduce the verbosity of 

[ 1948.701096] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
[ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x40070000: GLB_PM_MSG: CORE_ENABLE

I think the second message is good enough. We print both the success and error message when an IPC is sent.

Reply:
> @plbossart why do you say that? this is a snippet of what you'd see:
> [ 1948.700794] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30020000: GLB_TPLG_MSG: COMP_FREE
> [ 1948.701076] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x30020000: GLB_TPLG_MSG: COMP_FREE
> [ 1948.701096] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
> [ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x40070000: GLB_PM_MSG: CORE_ENABLE
> [ 1948.701422] sof-audio-pci-intel-tgl 0000:00:1f.3: core 3 powered down
> [ 1948.701429] sof-audio-pci-intel-tgl 0000:00:1f.3: widget SSP0.OUT freed
> 
> Without the above prints, all I know is something was either powered on/off but not what

sorry I read the patch sideways, it's only when you actually do something that you print the log, I thought it was at every call to put/get. no issue then

Reply:
> the recent Covid restrictions on time travel prevent you from knowing if an IPC is successful before deciding to print the first message - which is quite useful when you get a timeout, isn't it.

hmm right, we don't decode in case of a timeout. so I suppose we do need both.

Reply:
@ranj063, @plbossart, this exact duplicate prints for IPC tx bugs me as well. Atm we have prints
on success:
```
DBG [ 1948.701096] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
DBG [ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x40070000: GLB_PM_MSG: CORE_ENABLE
```
and on failure (the size is a guess):
```
DBG [ 1948.701096] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
ERR [ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx timed out for 0x40070000 (msg/reply size: 8/8)
```
We could reduce this to a single line for success and two lines for the failure paths by
on success:
```
DBG [ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x40070000: GLB_PM_MSG: CORE_ENABLE
```
and on fail (the size is a guess):
```
ERR [ 1948.701410] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx timed out for 0x40070000 (msg/reply size: 8/8)
DBG [ 1948.701096] sof-audio-pci-intel-tgl 0000:00:1f.3: failed message: 0x40070000: GLB_PM_MSG: CORE_ENABLE
```
Only printing in `tx_wait_done()`

Reply:
For debug it's interesting to see when an IPC was initiated. It'd be even more interesting if we had a joint kernel/firmware log with common timestamps to debug the entire platforms - something we've been planning for a long time but never worked on (debug tools never get the proper attention).

The success is less interesting really, it's expected. We could squelch the 'ipc tx succeeded' and only use additional information on failure.